src/goto-checker/solver_factory.cpp

```diff
@@ -241,6 +364,13 @@ std::unique_ptr<solver_factoryt::solvert> solver_factoryt::get_default()
   return solver;
 }
 
+void solver_factoryt::emit_solver_warning(const std::string &solver)
+{
+  messaget log(message_handler);
+  log.warning() << "The specified solver, '" << solver << "', is not available"
+                << messaget::eom;
```

Comment:
1) Could you please add a test for this case?
2) Perhaps the warning could also say that the default solver will be used instead.

Reply:
Done for both.